packages/suite/src/hooks/wallet/useStakeEthForm.ts

```diff
@@ -0,0 +1,332 @@
+import { saveComposedTransactionInfo } from 'src/actions/wallet/coinmarket/coinmarketCommonActions';
+import {
+    StakeEthFormState,
+    UseStakeEthFormProps,
+    StakeEthContextValues,
+    FIAT_INPUT,
+    CRYPTO_INPUT,
+    OUTPUT_AMOUNT,
+} from 'src/types/wallet/stakeEthForm';
+import {
+    getComposeAddressPlaceholder,
+    mapTestnetSymbol,
+} from 'src/utils/wallet/coinmarket/coinmarketUtils';
+import { useFormDraft } from './useFormDraft';
+import type { AppState } from 'src/types/suite';
+import { AmountLimits } from 'src/types/wallet/coinmarketCommonTypes';
+
+import { fromWei } from 'web3-utils';
+import { useStakeEthFormDefaultValues } from './useStakeEthFormDefaultValues';
+import { useCompose } from './form/useCompose';
+import { MIN_ETH_AMOUNT_FOR_STAKING, MIN_ETH_FOR_FEES } from 'src/constants/suite/ethStaking';
+import { selectLocalCurrency } from 'src/reducers/wallet/settingsReducer';
+
+export const StakeEthFormContext = createContext<StakeEthContextValues | null>(null);
+StakeEthFormContext.displayName = 'StakeEthFormContext';
+
+const useStakeEthState = (
+    selectedAccount: UseStakeEthFormProps['selectedAccount'],
+    fees: AppState['wallet']['fees'],
+    currentState: boolean,
+    defaultFormValues?: StakeEthFormState,
+) => {
+    // do not calculate if currentState is already set (prevent re-renders)
+    if (selectedAccount.status !== 'loaded' || currentState) return;
+
+    const { account, network } = selectedAccount;
+    const coinFees = fees[account.symbol];
+    const levels = getFeeLevels(account.networkType, coinFees);
+    const feeInfo = { ...coinFees, levels };
+
+    return {
+        account,
+        network,
+        feeInfo,
+        formValues: defaultFormValues,
+    };
+};
+
+export const useStakeEthForm = ({
+    selectedAccount,
+}: UseStakeEthFormProps): StakeEthContextValues => {
+    const dispatch = useDispatch();
+
+    const accounts = useSelector(state => state.wallet.accounts);
+    const device = useSelector(selectDevice);
+    const fiat = useSelector(state => state.wallet.fiat);
+    const localCurrency = useSelector(selectLocalCurrency);
+    const fees = useSelector(state => state.wallet.fees);
+
+    const { account, network } = selectedAccount;
+    const { symbol } = account;
+
+    const symbolForFiat = mapTestnetSymbol(symbol);
+    const fiatRates = fiat.coins.find(item => item.symbol === symbolForFiat);
+    // TODO: Implement fee switcher
+    const selectedFee = 'normal';
+
+    const amountLimits: AmountLimits = {
+        currency: symbol,
+        minCrypto: MIN_ETH_AMOUNT_FOR_STAKING.toNumber(),
+        maxCrypto: Number(account.formattedBalance),
+    };
+
+    const [state, setState] = useState<ReturnType<typeof useStakeEthState>>(undefined);
+
+    const { saveDraft, getDraft, removeDraft } = useFormDraft<StakeEthFormState>('stake-eth');
+    const draft = getDraft(account.key);
+    const isDraft = !!draft;
```

Comment:
Ok. But I thought about a couple of reasons to use saving drafts:

1. When you click "Continue", the "Confirm entry period" modal is opened. In the design it's visualized as open above the "Stake Ethereum" modal. But afaiu, in the app only one modal can be open at the same time. Opening another modal will lead to closing the previous one. So, if a user closes  the "Confirm entry period" modal, all input data will be lost and they will have to reopen the "Stake Ethereum" modal and put the values again. I thought to handle it by reopening the "Stake Ethereum" modal, when the user closes the "Confirm entry period" modal. And in such a case I can reopen it with all the values the user entered before.
2. I've also done it for consistency, since all the other forms do that.

If you're OK with stopping the flow when the user closes the "Confirm entry period" modal, then it makes sense to remove this.

Reply:
Added the "Clear all" button.
https://github.com/everstake/trezor-suite/pull/3/commits/4b797c81b4eeecab43293e84d3b2d83e841d0113

Reply:
nit: do not show button if form is empty

Reply:
https://github.com/everstake/trezor-suite/pull/3/commits/584c80d8901967315ef50144b377e9865927c791